![](./img/logo.png)

# 2do Parcial: Análisis Exploratorio de Datos. 

### Nombre de los Estudiantes:
    * David Steven Arbeláez Chávez
    * David Rodriguez
### Cédula de Ciudadanía:  
    * 1000 253 275

# Modelo **02**

### Descripción de Valores:

    join_date: fecha año-mes-dia hora:minuto:segundo.
    group: houston machine learning, houstonr, houston data science,
           houston energy data science, houston data visualization.
    city:  Houston, Katy, Orizaba, Spring, Cold Spring Harbor,
           Kenmore, Sugar Land, Colorado Springs, Pearland, Austin, ....
    state: TX, NY, WA, CO, MD, CA, LA, GA, MO, NC, FL, NH,
           MA, PA, KY, NE, IL, MN, TN, IN, OH, F2, NJ, NS, 
           VA, MI, 18, AR, AL, OR, DE, 17, ON, SC, BC, V5,
           WI, AB, DC, AK, UT, OK, AZ, MS, CT, IA.
    country: us, mx, il, be, cn, jp, gr, in, fr, my, de,
             ye, np, no, au, gb, dz, co, tw, br, ie, pk,
             ca, hk, nl, ng, ae, sg, eg, ir, es, nz, se,
             za, cz, th, ch, by, pe, fi, kr, si, ph.
         
    

## 0.- Bibliotecas y Lectura

In [19]:
import pandas as pd 
import numpy as np
import seaborn as sns
import matplotlib.pyplot as plt
from pandas import get_dummies

### _Se importa el data frame_ `meetup_groups `

In [20]:
df_meetup = pd.read_csv("./data/meetup_groups.csv")

## 1.- Identificación de datos perdidos (NaN)
    - Bonus --> ¿Cuantas Filas se han eliminado después de la limpieza?

### **Se identifica las columnas del dataframe**

In [21]:
df_meetup.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 7671 entries, 0 to 7670
Data columns (total 5 columns):
 #   Column     Non-Null Count  Dtype 
---  ------     --------------  ----- 
 0   join_date  7671 non-null   object
 1   group      7671 non-null   object
 2   city       7671 non-null   object
 3   state      7527 non-null   object
 4   country    7671 non-null   object
dtypes: object(5)
memory usage: 299.8+ KB


__Se muestran los valores unicos para identificar si hay valores nulos__

In [22]:
column_headers = list(df_meetup.columns.values)
for i in column_headers:
    
    a=str(i)
    print(f'Valores unicos de la columna {a} : \n    { (df_meetup[a].unique())}')
    print("***********************************************************************")
#df_meetup['state'].unique()


Valores unicos de la columna join_date : 
    ['2016-11-18 02:41:29' '2017-05-09 14:16:37' '2016-12-30 02:34:16' ...
 '2015-10-05 17:08:40' '2016-11-04 22:36:24' '2016-08-02 17:47:29']
***********************************************************************
Valores unicos de la columna group : 
    ['houston machine learning' 'houstonr' 'houston data science'
 'houston energy data science ' 'houston data visualization ']
***********************************************************************
Valores unicos de la columna city : 
    ['Houston' 'Katy' 'Orizaba' 'Spring' 'Cold Spring Harbor' 'Kenmore'
 'Sugar Land' 'Colorado Springs' 'Pearland' 'Austin' 'Tel Aviv-Yafo'
 'Baltimore' 'Garza García' 'Arlington' 'San Francisco' 'San Jose'
 'League City' 'Cypress' 'Stafford' 'Dallas' 'Friendswood' 'Lawtell'
 'Statesboro' 'North Houston' 'Brussels' 'Tomball' 'Beaumont' 'Bellaire'
 'Humble' 'Richmond' 'Port Lavaca' 'Montgomery' 'Rolla' 'Los Angeles'
 'Alameda' 'San Antonio' 'Bellevue' 'Durham' 'C

Se suma los nulos para poder identificarlos bien

In [23]:
df_meetup.isnull().sum()

join_date      0
group          0
city           0
state        144
country        0
dtype: int64

Tras realizar esta consulta de los datos unicos y tras revisar la información del data frame se puede afirmar el conjunto de datos tiene **datos Perdidos** en la columna `state` cuenta que con menos datos (los cuales fueron remplazados por datos nulos) que el resto de columnas, 144 datos menos.

Por lo que se va a quitar ese numero de columnas del resto de las columnas

In [24]:
#Se eliminan los datos nulos
df_meetup = df_meetup.dropna()

#Se verfica la informacion
df_meetup.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 7527 entries, 0 to 7670
Data columns (total 5 columns):
 #   Column     Non-Null Count  Dtype 
---  ------     --------------  ----- 
 0   join_date  7527 non-null   object
 1   group      7527 non-null   object
 2   city       7527 non-null   object
 3   state      7527 non-null   object
 4   country    7527 non-null   object
dtypes: object(5)
memory usage: 352.8+ KB


In [25]:
#Se verifica que se hayan quitado los datos nulos
df_meetup.isnull().sum()

join_date    0
group        0
city         0
state        0
country      0
dtype: int64

### Bonus

Se eliminaron 144 filas, debido a los datos nulos en la columna `state`

## 2.- ¿Cuantas reuniones por grupo se registraron en el DataFrame? 
    - Haga un gráfico (Pastel) de reuniones por grupo en el DataFrame

<img src="./img/grupos-meetings.png" style="width:250px">

In [ ]:
# Data to plot
labels = 'Python', 'C++', 'Ruby', 'Java'
labels = df_meetup[a].unique()
sizes = [215, 130, 245, 210]
explode = (0.1, 0, 0, 0)  # explode 1st slice

# Plot
plt.pie(sizes, explode=explode, labels=labels, autopct='%1.1f%%', shadow=True, startangle=140)

plt.axis('equal')
plt.show()

In [28]:
df_meetup.describe().T

count unique                   top  freq
join_date  7527   7487   2016-07-10 16:33:53    10
group      7527      5  houston data science  2150
city       7527    250               Houston  5560
state      7527     46                    TX  7118
country    7527      3                    us  7500

## 3.-  ¿Cuál es la proporción de reuniones en la ciudad Houston?

## 4.- Cuáles son las 10 ciudades más usadas en USA para reuniones de los grupos:
    - 'houston machine learning ' y 'houstonr'

## 5.- Cuáles son las ciudades más usadas para reuniones fuera del USA por:
    -  'houston machine learning' y 'houstonr'

## 6.- El  grupo 'houston data visualization' cuando hace reuniones en USA:
    - ¿Es cierto que la segunda ciudad en donde hacen más reuniones es Austin?

## 7.- ¿Cuál fue el mes que más reuniones tuvo el grupo 'houstonr' en el año 2016?

        - Agregar columna 'month' y 'year'

<img src="./img/graf02.png">

## 8.- ¿Cuantas reuniones por año se registraron en el DataFrame?

## 9.- ¿Cuantas veces por año se han reunido los grupos ('houston data science', 'houston machine learning') fuera de USA en el dataframe?